src/platform/forms-system/src/js/state/reducers.js

```diff
@@ -55,18 +55,27 @@ export default {
   }),
   [SET_SUBMISSION]: (state, action) => {
     const newState = _.set(['submission', action.field], action.value, state);
-    if (action.extra) {
-      newState.submission.extra = action.extra;
+    const { extra, errorMessage } = action;
+    const submission = {
+      ...newState.submission,
+      timestamp: new Date().getTime(),
+      hasAttemptedSubmit: true,
+    };
+    if (errorMessage) {
```

Comment:
This isn't related to the `submission` stuff, right?

Reply:
Yes, initially the plan was to only include `hasAttemptedSubmit`, but then seeing that there is a `timestamp` value pushed us to decide to include it, but instead of adding two separate values, we (Anna & I) discussed including the entire `submission` object in the SiPs data. The only problem was that the `timestamp` and `errorMessage` values are never updated. So I had to include those fixes.

Lastly, anticipating the error messages that will be added in https://github.com/department-of-veterans-affairs/vets-website/pull/14465, I included the `errors` block. 

We want to get a better handle on why a user would stop filling out the form on a page before the review & submit page. Did they stop without ever attempting to submit, or did they go back to try and fix a problem, but then gave up.